### PR TITLE
Moved convective top calculation to convection class

### PR DESCRIPTION
**Pull request by SallyDa**
and use the convective heating rates!

### CODE REVIEW

File: konrad/core.py [hunk 1]
```diff
@@ -229,24 +229,28 @@ def run(self):
             # TODO: Consider implementing an Atmosphere.update_diagnostics()
             #  method to include e.g. convective top in the output.
             self.atmosphere.update_height()
-            self.atmosphere.calculate_convective_top(
-                self.radiation['sw_htngrt_clr'][0, :] +
-                self.radiation['lw_htngrt_clr'][0, :])
+            z = self.atmosphere.get_values('z')[0, :]
+            if isinstance(self.convection, HardAdjustment) or isinstance(
+                    self.convection, RelaxedAdjustment):
+                self.convection.calculate_convective_top_height(z)
```
Comment (lkluft) [inline konrad/core.py]: Does this work for all subclasses of `Convection`? Then you could get rid of the check.

Comment (SallyDa) [reply konrad/core.py]: No, not for NonConvective (radiative equilibrium simulations), that's why I put the check there.

Comment (lkluft) [reply konrad/core.py]: One could consider to write a small implementation for the `NonConvective` case, that simply returns `None` or `nan` to keep the API of the classes consistent.

But we can also keep the check, as calculating a convective top for non-convective configurations might not be desirable 😉

Comment (SallyDa) [reply konrad/core.py]: We could do, but in the NonConvective case it seems a bit pointless to output convective heating arrays, as well as the convective top p, T and z, when there is no convective heating.

**Pull request merged**